include/mapbox/variant_cast.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef VARIANT_CAST_HPP
+#define VARIANT_CAST_HPP
+
+#include <type_traits>
+
+namespace mapbox { namespace util {
+
+namespace detail
+{
+
+template <class T>
+class static_caster
+{
+public:
+    template <class V>
+    T &operator()(V &v) const
+    {
+        return static_cast<T &>(v);
+    }
+};
+
+template <class T>
+class dynamic_caster
+{
+public:
+    using result_type = T&;
```

Comment:
Because i have 2 versions of dynamic cast - for references and for pointers, but one version for static cast. Constness in both cases is taken from the original object.